timemachine/cpp/src/log_flat_bottom_bond.cu

```diff
@@ -0,0 +1,79 @@
+#include "gpu_utils.cuh"
+#include "k_log_flat_bottom_bond.cuh"
+#include "kernel_utils.cuh"
+#include "log_flat_bottom_bond.hpp"
+#include "math_utils.cuh"
+#include <vector>
+
+namespace timemachine {
+
+template <typename RealType>
+LogFlatBottomBond<RealType>::LogFlatBottomBond(const std::vector<int> &bond_idxs, double beta)
+    : B_(bond_idxs.size() / 2), beta_(beta) {
+
+    // validate bond_idxs: even length, all idxs non-negative, and no self-edges
+    if (bond_idxs.size() % 2 != 0) {
+        throw std::runtime_error("bond_idxs.size() must be exactly 2*k!");
+    }
+
+    for (int b = 0; b < B_; b++) {
+        auto src = bond_idxs[b * 2 + 0];
+        auto dst = bond_idxs[b * 2 + 1];
+        if (src == dst) {
+            throw std::runtime_error("src == dst");
+        }
+
+        if ((src < 0) or (dst < 0)) {
+            throw std::runtime_error("idxs must be non-negative");
+        }
+    }
+
```

Comment:
should we also validate `beta`?

(assert non-negative, warn if not within 10% or so of room temperature in expected units?)

Reply:
Fair enough

(didn't mean to suggest an assertion for temperature, but just a warning. I think it is surprising for a potential energy function to accept `beta` (so it might be easier than other places to introduce unit errors or mistake what the parameter does). We use warnings elsewhere to detect possible temperature / pressure unit bugs

https://github.com/proteneer/timemachine/blob/88da6f3e57dc55b23f582787ba44f67599f6ee11/timemachine/cpp/src/barostat.cu#L28-L35
)

Reply:
> surprising for a potential energy function to accept beta

Sorry, to clarify: it is required and makes perfect sense for this function to accept `beta`, just it may be surprising to a user of the function (so may merit an extra warning).

(There's a note in the confluence page about whether we want to specify temperature in one place or another, I think the convention @badisa and I selected is preferable to the alternative, but neither alternative considered was perfect.)